Source/WebCore/html/ImageDocument.cpp

```diff
@@ -229,6 +230,13 @@ void ImageDocument::createDocumentStructure()
 
     auto body = HTMLBodyElement::create(*this);
     body->setAttribute(styleAttr, "margin: 0px");
+#if PLATFORM(HAIKU)
```

Comment:
This seems like the wrong way to do this. Isn't there a general "document colors" (aka, HTML3 colors) that we should be able to set at a higher level?

Reply:
Yes, or there should be.
In the other patch i have these two caces for css color variables CSSValueCanvas and CSSValueWindow. IIRC the gtk port also uses these two cases, but I have not gotten it to work correctly for a default canvas color. It is possible these are only set for some "this is the default color" css variables from css color module level 4
(https://www.w3.org/TR/css-color-4/#css-system-colors)
There a Canvas color name is defined.

I could probably try setting the background to Canvas and see if that helps.
(on another related note :root { color-scheme: light dark; } should be enough to change the document background color, but it does nothing with the color patch... I don't know why, the other css values seem to work)